radicle-cli/src/commands/remote.rs

```diff
@@ -0,0 +1,126 @@
+//! Remote Command implementation
+#[path = "remote/add.rs"]
+pub mod add;
+#[path = "remote/list.rs"]
+pub mod list;
+#[path = "remote/rm.rs"]
+pub mod rm;
+
+use std::ffi::OsString;
+
+use anyhow::anyhow;
+
+use radicle::prelude::Did;
+
+use crate::terminal as term;
+use crate::terminal::args::{string, Error};
+use crate::terminal::{Args, Context, Help};
+
+pub const HELP: Help = Help {
+    name: "remote",
+    description: "Manage set of tracked repositories",
+    version: env!("CARGO_PKG_VERSION"),
+    usage: r#"
+Usage
+    rad remote
+    rad remote list
+    rad remote add <url> [--alias]
+    rad remote rm <alias>
+Options
+        --help                 Print help
+"#,
+};
+
+#[derive(Debug, Default, PartialEq, Eq)]
+pub enum OperationName {
+    Add,
+    Rm,
+    #[default]
+    List,
+}
+
+#[derive(Debug)]
+pub enum Operation {
+    Add { did: Did, alias: Option<String> },
```

Comment:
We should make the Did be a 'Did' or an alias to lookup from the tracking database

Reply:
You can use an enum like we did in `assigned.rs` https://github.com/radicle-dev/heartwood/blob/master/radicle-cli/src/commands/issue.rs#L62

Reply:
Do you mean that the alias and the did should be merge in a single string?

Reply:
I am not sure that I got the suggestion

Reply:
I'll patch it, all good :+1:

Reply:
I do not understand how this will work, and I would like to understand it in the future, in the `rad issue` you can have that the issue assigned to yourself (it is easy to check in the code I guess?) or to a peer.

In the case of the Add, the `Did` is always present right? and then the alias will override the alias that the `rad` command assumes, so I do not know what this will look like, to me seems a little bit more complex? at least to read?

Reply:
Here's the [suggestion for using an enum](https://github.com/slack-coder/heartwood/commit/aa16c0cb413365c52ef32e763ef427f0e0f2f8ba).

Lets follow up with this change in another PR, and push ahead with this one as is.  I noticed some additional complexity.

 - tracking::Config lacks support for looking up a node by it's alias.
 - `rad remote rm` should support an `alias`, `git remote name`, and `did`.
 
 > In the case of the Add, the Did is always present right? and then the alias will override the alias that the rad command assumes, so I do not know what this will look like, to me seems a little bit more complex? at least to read?

 But it highlighted a few corrections for this PR.

 - We're mixing up word `alias` with git's remote `name`

    `rad remote add <url> [--alias]`

  Should be

    `rad remote add <did|nid> [--name]`
  
  Where the 'did' is the decentralized ID format for the Node Id (nid), and `name` will be what `git remote -v` will show.  Radicle has a relationship `alias -> Node ID` and git has a relationship `name -> URL`.  We should keep the terminology consistent not to confuse Radicle's aliases with Git remote names.

Reply:
Ah ok this is another feature, allow in the `remote add ` also a node id, yeah now make a lot of sense, thanks

Reply:
@vincenzopalazzo also `alias` -> `name` as `alias` is a Radicle concept, and `name` is git's remote name.